packages/react/dialog/src/Dialog.tsx

```diff
@@ -520,6 +525,23 @@ For more information, see https://radix-ui.com/primitives/docs/components/${labe
   return null;
 };
 
+const DESCRIPTION_WARNING_NAME = 'DialogDescriptionWarning';
+
+const DescriptionWarning: React.FC<WarningProps> = ({ contentRef }) => {
+  const descriptionWarningContext = useWarningContext(DESCRIPTION_WARNING_NAME);
+
+  const MESSAGE = `Warning: Missing \`Description\` or \`aria-describedby={undefined}\` for {${descriptionWarningContext.contentName}}.`;
+
+  React.useEffect(() => {
+    const hasDescription = document.getElementById(
+      contentRef.current?.getAttribute('aria-describedby')!
+    );
+    if (!hasDescription) console.warn(MESSAGE);
+  }, [MESSAGE, contentRef]);
```

Comment:
It looks like this would still warn if they passed `aria-describedby={undefined}`. Maybe need an extra check?

```suggestion
  React.useEffect(() => {
    const describedById = contentRef.current?.getAttribute('aria-describedby');
    const hasDescription = document.getElementById(describedById);
    if (describedById && !hasDescription) console.warn(MESSAGE);
  }, [MESSAGE, contentRef]);
```